index.js

```diff
@@ -284,6 +312,29 @@ module.exports = function (log, indexesPath) {
     }
   }
 
+  function addToPrefixMap(map, seq, value) {
+    if (value === 0) return
+
+    let arr = map[value] || []
+    arr.push(seq)
+    map[value] = arr
+  }
+
+  function updatePrefixMapIndex(opData, index, buffer, seq, offset) {
+    if (seq > index.count - 1) {
+      const fieldStart = opData.seek(buffer)
+      if (fieldStart) {
+        const buf = bipf.slice(buffer, fieldStart)
+        addToPrefixMap(index.map, seq, buf.length ? safeReadUint32(buf) : 0)
+      } else {
+        addToPrefixMap(index.map, seq, 0)
```

Comment:
I wonder what's the purpose of doing `map = {0: [seqA, seqB, seqC, seqD], ...}` when the `fieldStart` is zero? `fieldStart === 0` should mean that the field is at the very beginning of the `buffer`? @arj03

Reply:
Oh, and then `addToPrefixMap(index.map, seq, 0)` would actually *not* do anything because of line 315. I think this is somewhat confusing. Maybe we should do

```js
if (~fieldStart) {

}
```
with no `else`

Reply:
You mean the else block? Yeah, that's a bit wasteful, could just be removed. addToPrefixMap has a guard against 0 because that saves a ton of space in these kinds of indexes.

Reply:
Fixed